src/main/kotlin/me/zeroeightsix/kami/feature/module/DiscordStatus.kt

```diff
@@ -0,0 +1,100 @@
+import me.zeroeightsix.kami.event.TickEvent
+import net.arikia.dev.drpc.DiscordEventHandlers
+import net.arikia.dev.drpc.DiscordRPC
+import net.arikia.dev.drpc.DiscordRichPresence
+import net.arikia.dev.drpc.DiscordUser
+import net.minecraft.util.Util.OperatingSystem
+import net.minecraft.util.Util.getOperatingSystem
+
+@Module.Info(name = "DiscordRPC", category = Module.Category.MISC, description = "Discord Rich presence")
+object DiscordStatus : Module() {
+    @Setting(name = "Version")
+    private var version = true
+
+    @Setting(name = "Username")
+    private var username = true
+
+    @Setting(name = "Server")
+    private var server = true
+
+    @Setting(name = "Flair")
+    private var message = true
+
+    private const val updateLimit: Long = 15000L // 15 seconds, should be good. (prevents api rate limiting)
+    private const val applicationId = "753664640789118999"
+
+    val messages = arrayOf(
+            "kami red how???",
+            "its really here",
+            "bigrat.monster!!!",
+            "free and open source ;)",
+            "kami kami kami kami kami",
+            "Scythe goddard mass gleaning",
+            "i have a scoop!",
+            "Not unfounded.",
+            getOperatingSystemMessage(),
+            "YES!!!!!!",
+            "086 is NOT A PRIME NUMBER!",
+            "3 is a PRIME NUMBER",
+            "Also try our sister mod KAMI BLUE!",
+            ":) (smiley face)",
+            "Read rat ode by elizabeth acevedo"
+    )
+
+    var lastUpdate: Long = 0L
+
+    override fun onEnable() {
+        initDiscord()
+    }
+
+    override fun onDisable() {
+        DiscordRPC.discordShutdown()
+    }
+
+    @EventHandler
+    private val updateListener =
+            Listener<TickEvent.Client.InGame>({
+                if (lastUpdate == 0L) lastUpdate = System.currentTimeMillis()
+                if ((lastUpdate + updateLimit <= System.currentTimeMillis())) {
+                    // If boolean is true return value else return ""
+                    infix fun Boolean.to(string: String) = if (this) string else ""
+
+                    // This is the bottom half of the RPC with server ip and funny message
+                    val presence = DiscordRichPresence.Builder("${server to "127.0.0.1 (ez doxx)"}${(server && message) to " | "}${message to messages.random()}")
+                            // This is the top half with modver and username
+                            .setDetails("${username to "Notch"}${(username && version) to " | "}${version to KamiMod.MODVER}")
+                            // Image key (kami) and text when you scroll over it (kamiclient.com)
+                            .setBigImage("kami", "kamiclient.com")
+                            // Small image key and text when hovered over
+                            .setSmallImage("bigrat", "he is massive :)")
+
+                    //Build it
+                    DiscordRPC.discordUpdatePresence(presence.build())
```

Comment:
The significant action here is definitely updating the RPC, not building the information